### ISSUE TITLE
Editing a "live" connection doesn't apply new properties to that connection immediately

### ISSUE THREAD

**Issue by Charles-Gagnon** [microsoft/azuredatastudio#20985]
Version: 1.39.1 (user setup)
Commit: 7553f799e175f471b7590302dd65c997b838b29b
Date: 2022-08-30T01:11:13.047Z
VS Code: 1.62.0
Electron: 13.6.6
Chrome: 91.0.4472.164
Node.js: 14.16.0
V8: 9.1.269.39-electron.0
OS: Windows_NT x64 10.0.22621

1. Create a new connection to a server that doesn't have a valid server cert installed (don't change any properties yet)
2. Connection should succeed
3. While connection is active in OE right click on "Edit Connection" and change "Encrypt" to true in Advanced Properties
4. Click Connect
5. Connection will succeed (this is unexpected since Encrypt=true should mean the connection should fail)
6. Disconnect the connection in OE
7. Try to connect it again - it will fail with the expected error

So it seems like we aren't properly updating the "live" connection with the new properties when editing the connection.

![EditConnectionBug](https://user-images.githubusercontent.com/28519865/198105464-2ed2fd4d-94ee-4f4d-9a31-188ef253110b.gif)

**Comment by cheenamalhotra** [comment]
As discussed, and also mentioned here: https://github.com/microsoft/azuredatastudio/issues/20972#issuecomment-1291259140 the current design of connection profile identity is flawed and needs to be fixed. Profile identity creation needs to consider all connection properties (as applicable to provider) in order to make this work.

**Comment by kburtram** [comment]
@cheenamalhotra I think this can be fixed without changing the "connection identity" concept by forcing a reconnect when editing a connection.  In may make sense to decouple the two issues for implementation reasons.  Since this isn't a recent regression you can decide if you think this should be fixed for Nov.

**Comment by cheenamalhotra** [comment]
I looked into that but it seems reconnect does happen on the server tree, but with old advanced properties even though connection profile shows updated properties (which is not good news). I also looked in SQL Server profiler and saw new connection being made, but when I expect failure, it still manages to connect.

Dashboard does attempt the new options (if an existing dashboard for same connection is not already open), and if it fails with error appropriately. But if Dashboard is already open, it doesn't attempt to reconnect.
